python/lsst/meas/base/classification.py

```diff
@@ -0,0 +1,104 @@
+#!/usr/bin/env python
+#
+# LSST Data Management System
+# Copyright 2008-2016 AURA/LSST.
+#
+# This product includes software developed by the
+# LSST Project (http://www.lsst.org/).
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.    See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+#
+"""
+Definition and registration of classification plugins
+"""
+
+import numpy
+
+import lsst.pex.config
+from .afterburner import AfterburnerPluginConfig, AfterburnerPlugin
+from .pluginRegistry import register
+
+__all__ = (
+    "ABClassificationConfig", "ABClassificationPlugin",
+)
+
+
+class ABClassificationConfig(AfterburnerPluginConfig):
+
+    fluxRatio = lsst.pex.config.Field(dtype=float, default=.925, optional=True,
+                                      doc="critical ratio of model to psf flux")
+    modelErrFactor = lsst.pex.config.Field(dtype=float, default=0.0, optional=True,
+                                           doc="correction factor for modelFlux error")
+    psfErrFactor = lsst.pex.config.Field(dtype=float, default=0.0, optional=True,
+                                         doc="correction factor for psfFlux error")
+
+
+@register("base_ClassificationExtendedness")
+class ABClassificationPlugin(AfterburnerPlugin):
+    """
+    A binary measure of the extendedness of a source, based a simple cut on the ratio of the
+    PSF flux to the model flux.
+
+    Because the fluxes on which this algorithm is based are slot measurements, they can be provided
+    by different algorithms, and the "fluxRatio" threshold used by this algorithm should generally
+    be set differently for different algorithms.  To do this, plot the difference between the PSF
+    magnitude and the model magnitude vs. the PSF magnitude, and look for where the cloud of galaxies
+    begins.
+    """
+
+    ConfigClass = ABClassificationConfig
+
+    @classmethod
+    def getExecutionOrder(cls):
+        return cls.DEFAULT_AFTERBURNER
+
+    def __init__(self, config, name, schema, metadata):
+        AfterburnerPlugin.__init__(self, config, name, schema, metadata)
+        self.keyProbability = schema.addField(name + "_value", type="D",
```

Comment:
Well, for a while in SDSS we did explore how to make this a real float, and José Garmilla has code to convert these psf-model numbers into probabilities now (but we're not using it for LSST).

For now, let's keep it a float and describe it as a probability or likelihood.  @TallJimbo and I are not in favour of making it a bool/flag now as that really is an interpretation of a probability (i.e. involves priors).   If someone wants to start an RFC/c.l.o discussion of this (including adding a slot) that'd be fine, but I think it needs that level of discussion before we start changing things.